Index: pom.xml
===================================================================
--- pom.xml	(revision 11987)
+++ pom.xml	(working copy)
@@ -9,7 +9,7 @@
   </parent>
 
   <artifactId>findbugs-maven-plugin</artifactId>
-  <version>2.3.1</version>
+  <version>2.3.2-INTERNAL-1</version>
   <packaging>maven-plugin</packaging>
 
   <name>FindBugs Maven Plugin</name>
Index: src/main/groovy/org/codehaus/mojo/findbugs/FindBugsMojo.groovy
===================================================================
--- src/main/groovy/org/codehaus/mojo/findbugs/FindBugsMojo.groovy	(revision 11987)
+++ src/main/groovy/org/codehaus/mojo/findbugs/FindBugsMojo.groovy	(working copy)
@@ -519,9 +519,14 @@
       log.debug("XML outputFile is " + outputFile.getAbsolutePath())
       log.debug("XML output Directory is " + findbugsXmlOutputDirectory.getAbsolutePath())
 
+      ClassLoader origLoader = Thread.currentThread().getContextClassLoader();
+      try {
+        Thread.currentThread().setContextClassLoader( this.getClass().getClassLoader() );
+        executeFindbugs(locale, outputFile)
+      } finally {
+        Thread.currentThread().setContextClassLoader( origLoader );
+      }
 
-      executeFindbugs(locale, outputFile)
-
       if (!outputDirectory.exists()) {
         if ( !outputDirectory.mkdirs() ) {
           fail("Cannot create html output directory")
